### ISSUE TITLE
Version 1.0.28 gives "Unfortunately... has stopped. 

### ISSUE THREAD

**Issue by jhoc2000** [stovocor/whohasmystuff#30]
I updated from version 1.0.27 to 1.0.28 and when I try to run it I get the error "Unfortunately who has my stuff has stopped". 

I running it on a Sony Xperia Z2 with Android Marshmallow build 23.5.A.1.291.

**Comment by stovocor** [comment]
I am very sorry that this has happened to you. Thank you for reporting it here instead of just giving a 1-star "YOUR APP IS CRAP!" review. :)

This error showed up on my phone for the last few builds and went away after rebuilding the APK. So I tested the APK with my own phone before releasing it and I am surprised that the error still appears on different phones.

In order to solve this, I stopped using a deprecated compiler. Hopefully this will fix the issue once and for all.

The new version (1.0.29) has already been pushed to the Play Store and should be available in the next hours. Please let me know if this fixed your issue.

**Comment by jhoc2000** [comment]
Thank you for your prompt reply. 

When will version 1.0.29 be available on Aptoide? I don't have to Google account for privacy reasons and therefore can't use Google Play Store. 

I look forward to trying version 1.0.29. 

John.

**Comment by jhoc2000** [comment]
Alternatively, when will version 1.0.29 be available on F-droid? 

Thanks. 

John.

**Comment by stovocor** [comment]
> Alternatively, when will version 1.0.29 be available on F-droid?

I have no influence on the build process of F-Droid. Usually updates are very fast, 1.0.28 was released just yesterday on the Play Store.

However, since I had to make some changes to the build scripts, I do not know if the F-Droid guys have to make manual changes.

If you need it, I can provide you with the APK I signed for the Play Store in the meantime.

**Comment by jhoc2000** [comment]
Thank you very much for the offer but I'll wait for F-droid to host it. I'm not in that much of a hurry. 

John.

**Comment by stovocor** [comment]
Okay, that did not work out so well. I got at least one new crash report with the same error for 1.0.29.

Next try is 1.0.30, where I disabled ProGuard. Thanks for your patience.

**Comment by Roboe** [comment]
Just commenting here because I have the same reproducible bug (F-Droid version). I downgraded to 1.0.27. I had got a backup handy, thankfuly.
Can I provide you with some logcat? Running under Android 5.1 (Fairphone 2)

**Comment by stovocor** [comment]
> Just commenting here because I have the same reproducible bug (F-Droid version). I downgraded to 1.0.27. I had got a backup handy, thankfuly.

F-Droid still lists 1.0.28 as the current version and they did not even [notice](https://f-droid.org/wiki/page/Category:Apps_to_Update) that there is an update.

> Rejecting class de.freewarepoint.whohasmystuff.b because it failed compile-time verification (declaration of 'de.freewarepoint.whohasmystuff.b' appears in /data/app/de.freewarepoint.whohasmystuff-1/base.apk)

Thank you very much for the detailed log. I saw this error in the Google Play Console multiple times, but not with the detailed message you provided. However, "compile-time verification" is misleading since I did not get any error messages at compile time, at least for the version in the Play Store.

> Built F-Droid's 1.0.28 tarball just fine. To me it seems like a toolchain issue.

As stated above, the error vanished sometimes when I rebuilt the app. It looks like a race condition in ProGuard, though. Since I disabled the minification, there have been no more crash reports in the Google Play Console. So I am confident that version 1.0.30 is stable.

> P.S.: I'm having troubles syncing Gradle project on 1.0.29 and onwards because of signingConfigs

In order to eliminate the possibility that building with IntelliJ caused this issue, I moved to a standard Gradle toolchain. Because I do not want to add my key to the repo for obvious reasons, I followed a [setup mentioned on Stack Overflow](https://stackoverflow.com/questions/18328730/how-to-create-a-release-signed-apk-file-using-gradle). If you have any suggestion on how I can ease this process for you, please let me know.

**Comment by stovocor** [comment]
> If you have any suggestion on how I can ease this process for you, please let me know.

Okay, found it myself in the very same Stack Overflow thread. :)

Please check if you can build on the master branch now.

**Comment by Roboe** [comment]
F-Droid updated to 1.0.30 and it works now, :)

> In order to eliminate the possibility that building with IntelliJ caused this issue, I moved to a standard Gradle toolchain.

I'm confident F-Droid doesn't use IntelliJ at all in its build process, 😇 

> Please check if you can build on the master branch now.

Yeah! Love those conditionals, hehehe. Signing isn't needed at all for debug builds. Thanks!

**Comment by jhoc2000** [comment]
Hi, 

I have downloaded version 1.0.30 from F-droid, successfully installed it and ran it without any errors. I was also able to see all the books and to whom I have loaned them. 

Thank you very much for your prompt response. I'm so pleased with your obvious concern about getting the application working for your clientele. If only all application developers were as diligent as you. Well done. 

John.
